[2025-04-25T11:33:35.017+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-25T11:33:35.040+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_kafka_pipeline.run_spark_test scheduled__2025-04-25T11:10:00+00:00 [queued]>
[2025-04-25T11:33:35.051+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_kafka_pipeline.run_spark_test scheduled__2025-04-25T11:10:00+00:00 [queued]>
[2025-04-25T11:33:35.051+0000] {taskinstance.py:2306} INFO - Starting attempt 11 of 12
[2025-04-25T11:33:35.065+0000] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): run_spark_test> on 2025-04-25 11:10:00+00:00
[2025-04-25T11:33:35.071+0000] {standard_task_runner.py:63} INFO - Started process 228 to run task
[2025-04-25T11:33:35.074+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'stock_kafka_pipeline', 'run_spark_test', 'scheduled__2025-04-25T11:10:00+00:00', '--job-id', '400', '--raw', '--subdir', 'DAGS_FOLDER/test.py', '--cfg-path', '/tmp/tmp5zlfeqcn']
[2025-04-25T11:33:35.076+0000] {standard_task_runner.py:91} INFO - Job 400: Subtask run_spark_test
[2025-04-25T11:33:35.108+0000] {task_command.py:426} INFO - Running <TaskInstance: stock_kafka_pipeline.run_spark_test scheduled__2025-04-25T11:10:00+00:00 [running]> on host f4170a5e6dc6
[2025-04-25T11:33:35.166+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='stock_kafka_pipeline' AIRFLOW_CTX_TASK_ID='run_spark_test' AIRFLOW_CTX_EXECUTION_DATE='2025-04-25T11:10:00+00:00' AIRFLOW_CTX_TRY_NUMBER='11' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-25T11:10:00+00:00'
[2025-04-25T11:33:35.166+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-25T11:33:35.177+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-25T11:33:35.178+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'spark-submit --master spark://172.21.0.8:7077 /opt/airflow/spark/Scripts/test.py']
[2025-04-25T11:33:35.189+0000] {subprocess.py:86} INFO - Output:
[2025-04-25T11:33:36.963+0000] {subprocess.py:93} INFO - python3: can't open file '/opt/airflow/spark/Scripts/test.py': [Errno 2] No such file or directory
[2025-04-25T11:33:36.977+0000] {subprocess.py:93} INFO - 25/04/25 11:33:36 INFO ShutdownHookManager: Shutdown hook called
[2025-04-25T11:33:36.979+0000] {subprocess.py:93} INFO - 25/04/25 11:33:36 INFO ShutdownHookManager: Deleting directory /tmp/spark-a07aacc0-1106-4763-bc36-50bc7389797f
[2025-04-25T11:33:37.014+0000] {subprocess.py:97} INFO - Command exited with return code 2
[2025-04-25T11:33:37.014+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-25T11:33:37.020+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 2.
[2025-04-25T11:33:37.025+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=stock_kafka_pipeline, task_id=run_spark_test, run_id=scheduled__2025-04-25T11:10:00+00:00, execution_date=20250425T111000, start_date=20250425T113335, end_date=20250425T113337
[2025-04-25T11:33:37.035+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 400 for task run_spark_test (Bash command failed. The command returned a non-zero exit code 2.; 228)
[2025-04-25T11:33:37.051+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-04-25T11:33:37.064+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-25T11:33:37.065+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
